pkg/cmd/server/start/master_args.go

```diff
@@ -530,6 +541,18 @@ func (args MasterArgs) GetMasterAddress() (*url.URL, error) {
 	return url.Parse(masterAddr)
 }
 
+func (args MasterArgs) GetDNSDomain() (string, error) {
```

Comment:
Internally I would prefer that we use the standard subdomain (no trailing dot) to remove the possibility of confusion about whether it is required.  So if we find the suffix, we should strip it.  Most humans will observe the non-trailing dot version, and only the DNS server itself probably needs the dot.